### DISCUSSION TITLE
Custom plugin to block request and not send usage-reports

### DISCUSSION THREAD

**Discussion by flou** [apollographql/router#2715] [Q&A]
Hello,

I would like to create a custom plugin to add some validation, rules etc in order to block certain requests and I don't really want them in the usage reports.

Is there a way to block the request, respond with a status and a custom extension code, and prevent the router from sending the Apollo report (but still keep it enabled for other legit requests that I care about)?

**Comment by abernix** [comment]
Can you provide some more concrete examples of validations and rules you'd be able to provide?  And where would you be looking to write these customizations?  In Rhai?  Native Rust Extensions? Co-processors?

And also, why would you _not_ want observability on those operations in Studio? (Would there be some level of reporting you _would_ want to see on these?)
